### ISSUE TITLE
Coreeng: WAL buffer doesn't get flushed

### ISSUE THREAD

**Issue by filipkontic** [magley/nakevaleng#45]
Line 212 in put uses BufferedAppend to append the record to the WAL's buffer, however there isn't a FlushBuffer anywhere.
In places where we call the capital Put function, possibly many times (such as lines 267 and 275), we should make sure we flush the buffer.

**Comment by swopl** [comment]
Why?

**Comment by magley** [comment]
`BufferedAppend()` will call `Flush()` whenever the buffer is full, but we should also flush whenever a memtable is written to an SSTable, since those logs won't be needed anymore:
```go
func (cen CoreEngine) put(rec record.Record) {
        [...]
	cen.wal.BufferedAppend(rec)
	cen.cache.Set(rec)
	cen.sl.Write(rec)

	if cen.sl.Count > cen.conf.MemtableCapacity {
		[flush memtable, try compaction, ...]
                cen.wal.FlushBuffer() // <--- like this
		cen.wal.DeleteOldSegments()
	}
}
```

**Comment by swopl** [comment]
That one is OK, but why flush it in test()?

**Comment by magley** [comment]
I would only flush it before the program closes, if we want to save changes that aren't on disk, but then we'd need to implement recovery which is tricky unless low watermark is 0.

**Comment by swopl** [comment]
Fixed in [20aee59](https://github.com/magley/nakevaleng/pull/44/commits/20aee5934c6e8dd11dd64a394f3c8f9b9be9c9fc) and [504780b](https://github.com/magley/nakevaleng/pull/44/commits/504780b5d35bf9b2d51c97aae509f3f93d55a59c)
